src/v1/sources/shopify/webhookTransformations/serverSideUtlis.js

```diff
@@ -39,7 +39,25 @@ const createPropertiesForEcomEventFromWebhook = (message) => {
   return mappedPayload;
 };
 
+/**
+ * Returns the anonymousId from the noteAttributes array in the webhook event
+ * @param {Object} event
+ * @returns {String} anonymousId
+ */
+const getAnonymousIdFromAttributes = async (event) => {
+  let anonymousId = null;
+  const noteAttributes = event.note_attributes;
+  if (isDefinedAndNotNull(event) && isDefinedAndNotNull(noteAttributes)) {
```

Comment:
```suggestion
  if (isDefinedAndNotNull(event) && isDefinedAndNotNull(event.noteAttributes)) {
```
before the null check, note_attributes is being fetched. Move the noteAttributes initialisation after the check

Reply:
addressed